### PR TITLE
Add Static Activation Quantization Wrapper

### PR DESCRIPTION
**Pull request by namgyu-youn**
**Summary**:

- related to: #3655

Quantization tensor subclass (e.g., `Int8Tensor`, `Float8Tensor`) for static activation workflow requires a pre-computed activation scale (`act_quant_scale`), which means the activation should be updated before quantization. `act_quant_scale` can be computed by algorithms like AWQ/SmoothQuant. 

For example, AWQ workflow can be decomposed into:
```
(1) Prepare: Observer initialization
(2) Observe: Compute `act_quant_scale` based on algorithm and calibration flow (via `lm-eval`)
(3) Convert: Apply the `act_quant_scale` computed in (2)
(4) Quantize (RTN): Quantize using `quantize_(model, config)`
```

However, on the subclass (e.g., `Float8Tensor`, `Int8Tensor`) side, above workflow breaks subclass pattern, which should only address storage+inference ops. To handle this issue, this PR adds external wrapper to handle static activation workflow.

**Future Plan**:
- [ ] Update `Int8Tensor` to use this observer. (ref: https://github.com/pytorch/ao/pull/3687/changes/b373e101dafa670aa8d29841d0deb7d92f00d62e)
- [ ] Enum for custom use like AWQ
```py
class StaticObserver(Enum):
    """Observers supported to static quantization flow."""

    MINMAX = "affine"  # AffineQuantizedMinMaxObserver
    AWQ = "awq"  # AWQObserver
    SMOOTHQUANT = "smoothquant"  # SmoothQuantObserver
```
- [ ] Deprecate `Float8ObservedLinear` and update all callsite to use `ObserverLinear`
- [ ] Add NXFP format observer (see also: https://github.com/pytorch/ao/pull/3866#discussion_r2800398463)

**Test Plan**: CI

### CODE REVIEW

File: torchao/quantization/observer.py [hunk 1]
```diff
@@ -346,3 +346,68 @@ def calculate_qparams(self):
             self.preserve_zero,
             self.zero_point_domain,
         )
+
+
+# TODO: Build enum backend for more observers like AWQObserver, PerChannelHistogramObserver
+# User should be able to handle `act_obs` directly for easy customization
+class ObservedLinear(torch.nn.Linear):
```
Comment (jerryzh168) [inline torchao/quantization/observer.py]: is it possible to move this to https://github.com/pytorch/ao/tree/main/torchao/quantization/quantize_/common (create a new file like `observed_module.py`)

Comment (namgyu-youn) [reply torchao/quantization/observer.py]: Sure, moved to `torchao/quantization/quantize_/common/observer_module.py`

**Comment by jerryzh168** [comment]
@namgyu-youn we are actually thinking of move the ObservedLinear approach to use tensor subclass for observation: https://github.com/pytorch/ao/pull/3866#discussion_r2798336560

can you do that refactor first since you are working on this?

**Comment by jerryzh168** [comment]
current PR is a mix of refactor + int8 changes, so would be good to split anyways. I'd recommend to:
PR 1: do observed linear --> observed tensor refactor
PR 2: int8 changes

**Comment by namgyu-youn** [comment]
> @namgyu-youn we are actually thinking of move the ObservedLinear approach to use tensor subclass for observation: [#3866 (comment)](https://github.com/pytorch/ao/pull/3866#discussion_r2798336560)
> 
> can you do that refactor first since you are working on this?

Sure, small PR should always be better. Could you look into updated version?

**Comment by jerryzh168** [comment]
@namgyu-youn I think it would be good to just refactor this to use `ObservedTensor` instead of `ObservedLinear` instead as I mentioned before.

Also in terms of labels, we are moving to per module release: https://github.com/pytorch/ao/issues/3889, and we'll be using labels from https://github.com/pytorch/ao/labels?q=module going forward, this PR belongs to `module: inference` I think

**Comment by namgyu-youn** [comment]
@jerryzh168 renamed based on your suggestion, could you look into it?

**Comment by jerryzh168** [comment]
@namgyu-youn the issue is not about renaming, it's about the flow itself, what we want is:

prepared model:
  linear.weight = ObservedTensor(...)

converted model:
  linear.weight = QuantizedTensor(linear.weight, ...)

**Comment by namgyu-youn** [comment]
> @namgyu-youn the issue is not about renaming, it's about the flow itself, what we want is:
> 
> prepared model: linear.weight = ObservedTensor(...)
> converted model: linear.weight = QuantizedTensor(linear.weight, ...)

If I understood correctly, do you mean inherit subclass (like `Int8Tensor`, `Float8Tensor`) directly, not by linear module? So we can replace `ObservedTensor` with subclass at convert stage?

**Comment by jerryzh168** [comment]
> If I understood correctly, do you mean inherit subclass (li

yes that's correct

**Comment by namgyu-youn** [comment]
@jerryzh168 I think reland is better for fresh CI and updated context. Could you look into #3925 ?
